src/data-fetcher/bam/bam-worker.ts

```diff
@@ -335,50 +361,75 @@ function ChromosomeInfo(filepath, success) {
 /// End Chrominfo
 /////////////////////////////////////////////////////
 
-const bamRecordToJson = (bamRecord, chrName, chrOffset) => {
+const bamRecordToJson = (bamRecord: BamRecord, chrName: string, chrOffset: number) => {
     const seq = bamRecord.get('seq');
-    const segment = {
+    const segment: Segment = {
         // if two segments have the same name but different id, they are paired reads.
         // https://github.com/GMOD/bam-js/blob/7a57d24b6aef08a1366cca86ba5092254c7a7f56/src/bamFile.ts#L386
+        // @ts-expect-error private field!!
```

Comment:
Ah, I think we can later include this function inside the class to avoid the error.

Reply:
The `BamRecord` type coming from `@gmod/bam`, so it would need to be modified there. Perhaps there is a different method where we can get the `_id` without relying on a private field.

Reply:
TypeScript's `private` modifier is technically not actually private in runtime code, but trying to access that field means that we are relying on behavior that could change from the library without us knowing. I doubt it will, but something to keep in mind. Additionally, javascript recently added actual [private class fields](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/Private_class_fields), which if `gmod` ever used instead we couldn't rely on accessing the "private" fields.

Reply:
Perhaps `bamFile.seq_id` would be sufficent:

https://github.com/GMOD/bam-js/blob/9533ae737235b78876232cd923b4342b4de11987/src/record.ts#L49